### PR TITLE
Feat/update hooks

### PR DESCRIPTION
**Pull request by themonster2015**
## Description

* Replace all hooks that make use of react-query with an alternative

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/apps/notifications/src/components/pages/customize-notification-page.tsx [hunk 1]
```diff
@@ -178,18 +178,17 @@ const CustomizeNotificationPage: React.FC<CustomizeNotificationPageProps> = ({
     setLoading(true);
 
     try {
-      const allPrefs = allStates;
+      const allPrefs = structuredClone(allStates);
 
       //add snooze pref
       allPrefs[SnoozeOption] = snoozed;
 
       saveSettingsMutation.mutate(
```
Comment (SeverS) [inline ui/apps/notifications/src/components/pages/customize-notification-page.tsx]: use-settings hooks still using react-query

Comment (themonster2015) [reply ui/apps/notifications/src/components/pages/customize-notification-page.tsx]: Hi Sever! I think I removed react-query from it already :) https://github.com/AKASHAorg/akasha-core/pull/2200/files#diff-b427a501346a4e029682a4ca0c6767651953ab2ca72c6e20966daed1cf228490

**Comment by SeverS** [comment]
1. The general approach taken in this pr is to keep the interface (or at least partially) of the react-query useQuery and useMutation hooks. For me this is quite confusing. I would suggest to replace the functions like `mutate` to something else.
2. Some of the hooks have an overwhelming amounts of status flags (isLoading, isFetched, isSuccess, isError, etc.). This is because of the fact that it tries to keep the react-query interface. It will be harder and harder to maintain after some time. I would suggest to reduce the number of flags, or at least determine some of them from the state.

File: ui/apps/auth-app/src/components/connect/index.tsx [hunk 2]
```diff
@@ -19,7 +19,9 @@ const Connect: React.FC<unknown> = () => {
   const { data } = useGetLogin();
   const isLoggedIn = !!data?.id;
   const authenticatedDID = data?.id;
-  const logoutQuery = useLogout();
+
+  const loginCall = useLogin();
```
Comment (SeverS) [inline ui/apps/auth-app/src/components/connect/index.tsx]:
```suggestion
  const { signIn, signInErrors } = useLogin();
```

File: ui/apps/auth-app/src/components/connect/index.tsx [hunk 3]
```diff
@@ -19,7 +19,9 @@ const Connect: React.FC<unknown> = () => {
...
+  const logoutCall = useLogout();
```
Comment (SeverS) [inline ui/apps/auth-app/src/components/connect/index.tsx]:
```suggestion
  const {logOut, logOutErrors} = useLogout();
```

File: ui/apps/auth-app/src/components/connect/index.tsx [hunk 4]
```diff
@@ -60,15 +60,15 @@ const Connect: React.FC<unknown> = () => {
   }, [isLoggedIn, loading, authenticatedDID, profileData, profileDataReq, worldConfig.homepageApp]);
 
   const handleDisconnect = () => {
-    logoutQuery.mutate();
+    logoutCall.logOut();
```
Comment (SeverS) [inline ui/apps/auth-app/src/components/connect/index.tsx]:
```suggestion
    logOut();
```

File: ui/apps/auth-app/src/components/connect/index.tsx [hunk 5]
```diff
@@ -60,15 +60,15 @@ const Connect: React.FC<unknown> = () => {
...
     routingPlugin.current?.navigateTo({
       appName: '@akashaorg/app-auth-ewa',
       getNavigationUrl: appRoutes => `${appRoutes[CONNECT]}`,
     });
   };
 
   const handleSignIn = provider => {
-    loginMutation.mutate({ selectedProvider: provider });
+    loginCall.signIn({ selectedProvider: provider });
```
Comment (SeverS) [inline ui/apps/auth-app/src/components/connect/index.tsx]:
```suggestion
    signIn({ selectedProvider: provider });
```

File: ui/apps/auth-app/src/components/connect/index.tsx [hunk 6]
```diff
@@ -84,7 +84,7 @@ const Connect: React.FC<unknown> = () => {
               onSignIn={handleSignIn}
               onDisconnect={handleDisconnect}
               worldName={worldConfig.title}
-              signInError={loginMutation.error}
+              signInError={loginCall.error}
```
Comment (SeverS) [inline ui/apps/auth-app/src/components/connect/index.tsx]:
```suggestion
              signInError={signInErrors}
```

File: ui/hooks/src/use-login.new.ts [hunk 7]
```diff
@@ -111,29 +131,54 @@ export const useGetLoginProfile = () => {
...
         }
-      },
+      };
+
+      signInApiCall();
     },
+    [],
   );
+
+  return { signIn, data, isLoading, error, isSuccess: !!data, isError: !!error };
```
Comment (SeverS) [inline ui/hooks/src/use-login.new.ts]:
```suggestion
  return { signIn, data, isLoading, signInErrors, isSuccess: !!data, isError: !!error };
```

File: ui/hooks/src/use-login.new.ts [hunk 8]
```diff
@@ -150,16 +195,30 @@ export function useLogin(onError?: (err: Error) => void) {
...
-    },
-    {
-      onError: (err: Error) => logError('use-logout', err),
-    },
-  );
+    };
+
+    logoutApiCall();
+
+    return { data, isLoading, error };
```
Comment (SeverS) [inline ui/hooks/src/use-login.new.ts]: you are trying to return the error in logout function via the state. Note that the state might not be updated yet. Maybe you want to just return the data? and return the error via the hook?

File: ui/hooks/src/use-notifications.ts [hunk 9]
```diff
@@ -1,59 +1,45 @@
...
-      logError('useNotifications.markAsRead', err as Error);
-    },
-    onSettled: async () => {
-      await queryClient.invalidateQueries([NOTIFICATIONS_KEY]);
-    },
-  });
+    };
+
+    markMessageAsReadApiCall();
+    return { data, isLoading, error, isSuccess: !!data, isError: !!error };
```
Comment (SeverS) [inline ui/hooks/src/use-notifications.ts]: same as in the logout function

File: ui/hooks/src/use-settings.ts [hunk 10]
```diff
@@ -38,23 +35,37 @@ async function saveSettings({
...
+            callback.onComplete();
+          }
+        } catch (err) {
+          setError(err);
+        }
+      };
+
+      saveSettingCall(params);
+
+      return { isLoading, data, error };
```
Comment (SeverS) [inline ui/hooks/src/use-settings.ts]: same issue here!

File: ui/widgets/sidebar/src/components/sidebar-component.tsx [hunk 11]
```diff
@@ -50,8 +46,7 @@ const SidebarComponent: React.FC<unknown> = () => {
   const [isLoading, setIsLoading] = useState(false);
   const authenticatedDID = data?.id;
   const isLoggedIn = !!data?.id;
-  const logoutMutation = useLogout();
-  const queryClient = useQueryClient();
+  const logoutCall = useLogout();
```
Comment (SeverS) [inline ui/widgets/sidebar/src/components/sidebar-component.tsx]:
```suggestion
  const { logOut } = useLogout();
```

File: ui/widgets/sidebar/src/components/sidebar-component.tsx [hunk 12]
```diff
@@ -181,9 +176,7 @@ const SidebarComponent: React.FC<unknown> = () => {
 
   function handleLogout() {
     setIsLoading(true);
-
-    queryClient.setQueryData([LOGIN_STATE_KEY], null);
-    logoutMutation.mutate();
+    logoutCall.logOut();
```
Comment (SeverS) [inline ui/widgets/sidebar/src/components/sidebar-component.tsx]:
```suggestion
    logOut();
```

**Review by SeverS — APPROVED** [review]

**Review by kenshyx — COMMENTED** [review]
[vandamme](https://app.graphite.dev/settings/meme-library?org=AKASHAorg)

**Pull request merged**